### ISSUE TITLE
PSS signature fails with legacy RSA_METHOD

### ISSUE THREAD

**Issue by alonbl** [openssl/openssl#21008]
Hello,

Root casue: [#19724](https://github.com/openssl/openssl/pull/19724)
`master`: 5a3bbe1712435d577bbc5ec046906979e8471d8b
`>=openssl-3.0.8`
`>=openssl-3.1.0`

Previous behavior calculated the RSA PSS salt length explicitly.
Current behavior queries the provider using `EVP_PKEY_CTX_get_params()`.

Legacy `RSA_METHOD` replacement based on `RSA_get_default_method()` fails as the call to `EVP_PKEY_CTX_get_params()` fails as it does not implement command handling and revert to the default check which rejects any query.

```
0062E2B41B7F0000:error:03000093:digital envelope routines:default_check:command not supported:crypto/evp/ctrl_params_translate.c:330:
```

I am unsure if this breakage is intentional or not, I understand that it is difficult to provide full backward compatibility.
In openssl-3 providers it still much more complex to override only rsa enc/dec and enjoy the rest of the functionality, it is not trivial to migrate existing code.

Please tell me if you need additional information.
If this is intentional behavior can you think of a workaround? I could not find anything simple.

Regards,

**Comment by alonbl** [comment]
@neverpanic: What do you think?

**Comment by t8m** [comment]
Fix in #21019. Please test it.
